x-pack/plugins/security/server/authentication/api_key.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export interface CreateApiKeyOptions {
+  callCluster: any;
+  isSecurityFeatureDisabled: () => boolean;
+  body: {
+    name: string;
+    role_descriptors: Record<string, any>;
+    expiration?: string;
+  };
+}
+
+export interface CreateApiKeyResult {
```

Comment:
__nit:__ can you please document this interface and its properties a bit? So that future readers/maintainers can quickly get the idea and go to the documentation web site only when more details are needed?